libraries/base/Data/Functor.hs

```diff
@@ -11,8 +11,29 @@
 -- Stability   :  provisional
 -- Portability :  portable
 --
--- Functors: uniform action over a parameterized type, generalizing the
--- 'Data.List.map' function on lists.
+--
+-- A type @f@ is a Functor if it provides a function @fmap@ which, given any types @a@ and @b@,
+-- lets you apply any function of type @(a -> b)@ to turn an @f a@ into an @f b@, preserving the
+-- structure of @f@.
+--
+-- ==== __Examples__
+--
+--  >>> fmap show (Just 1)  -- fmap :: (a   -> b)      -> f a       -> f b
+--  Just "1"                -- fmap :: (Int -> String) -> Maybe Int -> Maybe String
+--
+--  >>> fmap show Nothing   -- fmap :: (a   -> b)      -> f a       -> f b
+--  Nothing                 -- fmap :: (Int -> String) -> Maybe Int -> Maybe String
+--
+--  >>> fmap show [1,2,3]   -- fmap :: (a   -> b)      -> f a       -> f b
+--  ["1", "2", "3"]         -- fmap :: (Int -> String) -> [Int]     -> [String]
+--
+--  >>> fmap show []        -- fmap :: (a   -> b)      -> f a       -> f b
+--  []                      -- fmap :: (Int -> String) -> [Int]     -> [String]
+--
+-- The 'fmap' function is also available as the infix operator '<$>':
+--
+--  >>> show <$> (Just 1)  -- fmap :: (Int -> String) -> Maybe Int -> Maybe String
```

Comment:
I wonder whether the parens are helping or hurting here. I've seen newcomers have some trouble grasping Haskell's precedence scheme; I fear that adding extraneous parens, which making this example more explicit, may contribute to this confusion.

Reply:
Hm, I only ever found the parens helpful myself, especially as a beginner. Also it’s consistent with the actual fmap implementation. I would really like to keep them unless this is a dealbreaker for you or anyone else.

Reply:
I wonder if we could show a set of examples and talk about the precedence?

```
λ> show <$> Just 1
Just "1"
it :: Maybe String

λ> show <$> (Just 1)
Just "1"
it :: Maybe String

λ> fmap show Just 1
"Just 1"
it :: String

λ> fmap show (Just 1)
Just "1"
it :: Maybe String
```
I think that newcomers are probably surprised by the difference in behavior of the prefix vs infix versions of `fmap` as above. Maybe we could explain it and/or provide the examples instead of simply giving the `infixl` without comment? Precdence does take a lot of getting used to.

Reply:
Yes and no - while my PRs are all about making the docs as approachable and "user friendly" as possible i am unsure about this. If we elaborate on this we are not strictly talking about Functor anymore we are discussing how precedence works - and then this might as well be placed anywhere else in the documentation of any other module, or all of them, or none of them?

:thinking:

Reply:
```
--  >>> fmap show (Just 1) --  (Int -> String) -> Maybe Int -> Maybe String
--  Just "1"
--  >>> show <$> (Just 1)  --  (Int -> String) -> Maybe Int -> Maybe String
--  Just "1"
```

not sure if this is perhaps some sort of compromise - illustrating that the two are equivalent.

Reply:
Yeah, that's fair -- there probably isn't a good place for this, which may be why it remains confusing. The examples are, at any rate, specifically about the infix operator `<$>` and how it differs from the prefix `fmap`.